websites/M/Musixmatch/presence.ts

```diff
@@ -1,45 +1,106 @@
-	else if (path("/lyrics/")) {
-		presenceData.details = document
-			.querySelector("#site h1")
-			.textContent.replace("Lyrics", "");
-		presenceData.state = document.querySelector(
-			"#site h2 > span > a"
-		).textContent;
-		presenceData.smallImageKey = "reading";
-	} else if (path("/community")) presenceData.details = "Viewing community";
-	else if (path("/profile/")) {
-		presenceData.details = "Viewing the profile:";
-		presenceData.state = document.querySelector("h1").textContent;
-	} else if (path("/artist/")) {
-		presenceData.details = "Viewing an artist:";
-		presenceData.state = document.querySelector("#content h1").textContent;
-	} else if (path("/search/")) {
-		presenceData.details = "Searching";
-		presenceData.state = document.querySelector(
-			"#content span > span"
-		).textContent;
-		presenceData.smallImageKey = "search";
+	if (domain === "musixmatch.com" || domain === "www.musixmatch.com") {
+		if (path === "/") presenceData.details = "Viewing the home page";
+		else if (path.includes("/explore"))
+			presenceData.details = "Exploring lyrics";
+		else if (path.includes("/community"))
+			presenceData.details = "Viewing the community page";
+		else if (path.includes("/contribute"))
+			presenceData.details = "Viewing the contribute page";
+		else if (path.includes("/profile/me")) {
+			//if user status = curator show curator badge
+			if (
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > span"
+				)
+			) {
+				presenceData.smallImageKey = "https://i.imgur.com/Bl7BSIW.png";
+				presenceData.smallImageText = "Curator";
+			}
+			presenceData.details = "Viewing their profile";
+		} else if (path.includes("/profile/")) {
+			//if user status = curator show curator badge
+			if (
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > span"
+				)
+			) {
+				presenceData.smallImageKey = "https://i.imgur.com/Bl7BSIW.png";
+				presenceData.smallImageText = "Curator";
+			}
+			presenceData.details = `Viewing ${
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > h1"
+				).textContent
+			}'s profile`;
+		} else if (path.includes("/search")) {
+			presenceData.details = "Searching";
+			presenceData.state = path.split("/")[2].replaceAll("%20", " ");
+		} else if (path.includes("/lyrics") && path.includes("/translation/")) {
+			presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+				"#site > div > div > div > main > div > div > div.mxm-track-banner.top > div > div > div > div.col-sm-1.col-md-2.col-ml-3.col-lg-3.static-position > div > div > div > img"
+			).src;
```

Comment:
Better selectors e.g.
```suggestion
			presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
				'[property="og:image"]').content;
```